From: Pablo Lorenzzoni <spectra@debian.org>
Date: Wed, 5 Feb 2014 16:57:49 -0200
Subject: Fix cppcheck bugs #8 e #9

[dgate.cpp:5985]: (error) Uninitialized variable: f
[dgate.cpp:6422]: (error) Uninitialized variable: owned
---
 dgate.cpp |   13 ++++++-------
 1 file changed, 6 insertions(+), 7 deletions(-)

diff --git a/dgate.cpp b/dgate.cpp
index ff462b2..ea22a30 100644
--- a/dgate.cpp
+++ b/dgate.cpp
@@ -5983,13 +5983,12 @@ extern "C"
     { f = DB.Exec(sql);
       DB.Close();
       QueryFromGui++;
+      if (f)
+      { lua_pushinteger(L, 1);
+        return 1;
+      }
     }
-    if (f)
-    { lua_pushinteger(L, 1);
-      return 1;
-    }
-    else
-      return 0;
+    return 0;
   }
   static int luadicomquery(lua_State *L)
   { const char *ae    = lua_tostring(L,1);
@@ -6410,7 +6409,7 @@ extern "C"
   static int luadeletedicomobject(lua_State *L)
   { struct scriptdata *sd = getsd(L);
     DICOMDataObject *O = NULL;
-    int owned;
+    int owned = 0;
     Array < DICOMDataObject * > *A = NULL;
     if (lua_isuserdata(L,1)) 
     { lua_getmetatable(L, 1);
